api/src/org/labkey/api/security/EntropyPasswordValidator.java

```diff
@@ -54,7 +54,9 @@ public boolean shouldShowPasswordGuidance()
 
     public static double score(String password, User user)
     {
-        return calculateEntropy(filter(password.toCharArray(), user));
+        // Score only the first 100 characters, Issue 53814
+        String shorter = password.substring(0, 100);
```

Comment:
This throws StringOutOfBoundsException when the length < 100 chars.


```suggestion
        String shorter = password.substring(0, Math.min(password.length(), 100));
```

Reply:
You don't want to encourage long (> 100 char) passwords? Well, okay.